packages/ai-bot/helpers.ts

```diff
@@ -283,40 +288,59 @@ export function getModifyPrompt(
     let body = event.content.body;
     if (body) {
       if (event.sender === aiBotUserId) {
-        historicalMessages.push({
-          role: 'assistant',
-          content: body,
-        });
+        let functionCall = event.content.data?.functionCall;
+        if (functionCall) {
+          console.log('functionCall: ', functionCall); // Don't add to message list, need a result otherwise there's an API error
+        } else {
+          historicalMessages.push({
+            role: 'assistant',
+            content: body,
+          });
+        }
       } else {
-        historicalMessages.push({
-          role: 'user',
-          content: body,
-        });
+        // With a tool result we must also construct the message from the assistant with the tool call in it.
```

Comment:
To return the result of a tool call to openai, you have to include the message that had the call in as well.

However, if you do not have a result, having this tool call in the history causes an API error. So we need to ignore it *unless* there is a result, and only then add it to the conversation.

Reply:
It is also worth removing the matrix body (where we put a human readable message) otherwise after a while it starts to try and copy this itself, rather than actually calling the required tool.